integration_tests/src/main/python/data_gen.py

```diff
@@ -906,6 +906,7 @@ def gen_scalars_for_sql(data_gen, count, seed=0, force_no_nulls=False):
 
 # Some array gens, but not all because of nesting
 array_gens_sample = single_level_array_gens + nested_array_gens_sample
+array_of_map_gen = ArrayGen(MapGen(StringGen(pattern='key_[0-9]', nullable=False), StringGen(), max_length=10), max_length=10)
```

Comment:
nit: this name is a little generic. Nested types makes it difficult to name things well. Perhaps `array_map_string_string_gen`. It is ugly but there is less ambiguity. Also if this only used in one file, or one place in the code we don't need to create it in data_gen.py

Reply:
For now, this is only used in `array_test`.

Reply:
updated